packages/flutter_blue_plus_platform_interface/lib/src/bluetooth_msgs.dart

```diff
@@ -300,15 +300,16 @@ class BmBluetoothCharacteristic {
   final DeviceIdentifier remoteId;
   final Guid serviceUuid;
   final Guid characteristicUuid;
+  final int characteristicIndex;
   final Guid? primaryServiceUuid;
   List<BmBluetoothDescriptor> descriptors;
   BmCharacteristicProperties properties;
 
-
   BmBluetoothCharacteristic({
     required this.remoteId,
     required this.serviceUuid,
     required this.characteristicUuid,
+    required this.characteristicIndex,
```

Comment:
Should we default this to `0` and make it optional in order to avoid a breaking change?

```suggestion
    this.characteristicIndex = 0,
```

https://github.com/chipweinberger/flutter_blue_plus/issues/795#issuecomment-2461566048